File: modules/stego-core/src/main/java/com/shterneregen/securelan/stego/model/BmpCapacity.java
package com.shterneregen.securelan.stego.model;

public record BmpCapacity(
        int width,
        int height,
        int bitsPerPixel,
        int carrierBytes,
        int headerBytes,
        int payloadCapacityBytes
) {
    public BmpCapacity {
        if (width <= 0) {
            throw new IllegalArgumentException("width must be positive");
        }
        if (height <= 0) {
            throw new IllegalArgumentException("height must be positive");
        }
        if (bitsPerPixel != 24 && bitsPerPixel != 32) {
            throw new IllegalArgumentException("bitsPerPixel must be 24 or 32");
        }
        if (carrierBytes < 0) {
            throw new IllegalArgumentException("carrierBytes must not be negative");
        }
        if (headerBytes <= 0) {
            throw new IllegalArgumentException("headerBytes must be positive");
        }
        if (payloadCapacityBytes < 0) {
            throw new IllegalArgumentException("payloadCapacityBytes must not be negative");
        }
    }
}

File: modules/stego-core/src/main/kotlin/com/shterneregen/securelan/stego/model/BmpCapacity.kt
package com.shterneregen.securelan.stego.model

@JvmRecord
data class BmpCapacity(
    val width: Int,
    val height: Int,
    val bitsPerPixel: Int,
    val carrierBytes: Int,
    val headerBytes: Int,
    val payloadCapacityBytes: Int,
) {
    init {
        require(width > 0) { "width must be positive" }
        require(height > 0) { "height must be positive" }
        require(bitsPerPixel == 24 || bitsPerPixel == 32) { "bitsPerPixel must be 24 or 32" }
        require(carrierBytes >= 0) { "carrierBytes must not be negative" }
        require(headerBytes > 0) { "headerBytes must be positive" }
        require(payloadCapacityBytes >= 0) { "payloadCapacityBytes must not be negative" }
    }
}
